src/gui/dragAndDrop.js

```diff
@@ -35,16 +37,14 @@ function dropNode(event) {
     }
 
     Calculator.updateResult();
-  // Add node to tree/forest here
 }
 
 function deleteNode(event) {
     event.preventDefault();
     const id = event.dataTransfer.getData("text");
     document.getElementById(id).remove();
+    NodeForest.deleteRootNode(nodeCount);
```

Comment:
deleteRootNode should use the node ID, which should be equal to the HTML ID. This ID is generated when Node is instantiated and returned by NodeFactory.createNode().